benchmarks/encoder/utils.py

```diff
@@ -47,6 +48,8 @@ def gen_traces(config):
             resource = random.choice(resources)
             service = random.choice(services)
             with Span(None, span_name, resource=resource, service=service, parent_id=parent_id) as span:
+                if i == 0 and config.dd_origin:
```

Comment:
maybe `if span.context.dd_origin != config.dd_origin:` ?

Reply:
what's the reasoning for this? Not arguing, just thought we only need to set it on the root span anyway, and `config.dd_origin` is just the boolean indicator that `dd_origin` should be set

Reply:
we aren't setting it on the root, we are setting it on the context, so we just need to be sure it has it at some point before the trace finishes.

being the first, fifth, or last span doesn't actually matter.

Reply:
I actually didn't realize that this util method `gen_traces()` uses the `Span(...)` API which doesn't have the automatic propagation of contexts between parent and child spans as compared to using a tracer. In this case, being the first span actually does matter 😞 

This means we can:
1) change this method entirely to use the Tracer API, or 
2) we set the dd_origin once on the root's context, which is the only span's context checked by the encoder anyway in `_pack_trace()` before propagating the dd_origin to individual spans in the trace.

What do you think?

Reply:
Ah!

then what you have here makes sense, but let's add a comment!